docs/index.rst

```diff
@@ -22,7 +22,7 @@ It enables both the binding of data to a map for choropleth visualizations
 as well as passing rich vector/raster/HTML visualizations as markers on the map.
 
 The library has a number of built-in tilesets from OpenStreetMap,
-Mapbox, and Stamen, and supports custom tilesets.
+Mapbox, etc, and supports custom tilesets.
```

Comment:
@Conengmo I wonder if we should leave OSM as default and remove everything else in lieu of `xyzservices`.

Reply:
Or we could make `xyzservices` a required dependency (pure python, no dependencies) and offload all of it, while giving users all available tiles by default.

Reply:
That is kind of what I was thinking. Making is a mandatory dependency using it everywhere instead of having the templates for CartoDB, which was the last one to survive the API key revolution ;-p

Reply:
Then we agree :). If we want to do that, we should expose `xyzservices` via the `query_name` method as we do in `geopandas` or `contextily`, so you don't need to interact with `xyzservices` directly.

Reply:
What do you think @Conengmo ? Should I start that in this PR?

Reply:
Really interesting thought! Can you give me a moment? I don’t want to rush my thoughts here. I’ll get back this weekend if that’s okay.

Reply:
We are not in a hurry 😄 
But b/c this is kind of a breaking change I would like to hold the next release to ship this with it, if we decide to implement it.

PS: the devil is in the details, we have tons of examples and docs that will request some re-writing. However, the disruption on end user won't be too big, I hope.

Reply:
> we have tons of examples and docs that will request some re-writing. However, the disruption on end user won't be too big, I hope.

Why? Apart from Stamen maps not being available, which is nothing we can influence, it should be completely backwards compatible. Can you point me to examples that would need re-writing?

Reply:
I'm on mobile now to check but my guess are the names we use in the tile keywords in the do docstring and examples. Those would change to match XYZ, no? I'd love for us to use the same as geopandas to reduce the cognitive load on users

Reply:
If exposed via `query_name` they should, with the exception of "OpenStreetMap", which needs to be mapped to "OpenStreetMap.Mapnik" but we can sort this one case internally I suppose.